artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -528,8 +543,17 @@ public void decodeHeadersAndProperties(final ByteBuf buffer) {
    private void decodeHeadersAndProperties(final ByteBuf buffer, boolean lazyProperties) {
       messageIDPosition = buffer.readerIndex();
       messageID = buffer.readLong();
-
-      address = SimpleString.readNullableSimpleString(buffer);
+      int b = buffer.readByte();
```

Comment:
Shouldnt this logic remain in SimpleString, as you could read strings (especially address in many places)